File: reto17/reto17.py
def detect_bombs(grid: list[list[bool]]) -> list[list[int]]:
    rows, cols = len(grid), len(grid[0])
    result = [[0] * cols for _ in range(rows)]

    neighbors = [
        (-1, 0),
        (-1, 1),
        (0, 1),
        (1, 1),
        (1, 0),
        (1, -1),
        (0, -1),
        (-1, -1),
    ]

    for i in range(rows):
        for j in range(cols):
            result[i][j] = sum(
                grid[i + di][j + dj]
                for di, dj in neighbors
                if 0 <= i + di < rows and 0 <= j + dj < cols
            )

    return result

File: reto17/reto17.js
/**
 * @param {boolean[][]} grid
 * @returns {number[][]}
 */
function detectBombs(grid) {
  const result = [];
  const parseValue = {
    true: 1,
    false: 0,
    undefined: 0,
  };
  for (let i = 0; i < grid.length; i++) {
    for (let j = 0; j < grid[0].length; j++) {
      result[i] = result[i] || [];
      const ySide = parseValue[grid[i - 1]?.[j]] + parseValue[grid[i + 1]?.[j]];
      const xSide = parseValue[grid[i]?.[j - 1]] + parseValue[grid[i]?.[j + 1]];
      const upperDiag =
        parseValue[grid[i - 1]?.[j - 1]] + parseValue[grid[i - 1]?.[j + 1]];
      const lowerDiag =
        parseValue[grid[i + 1]?.[j - 1]] + parseValue[grid[i + 1]?.[j + 1]];
      result[i][j] = ySide + xSide + upperDiag + lowerDiag;
    }
  }
  return result;
}
